apisix/cli/file.lua

```diff
@@ -293,7 +294,9 @@ function _M.read_yaml_conf(apisix_home)
         end
     end
 
-    if default_conf.deployment.config_provider == "yaml" then
+    --- using `not ngx` to check whether the current execution environment is apisix cli module,
+    --- because it is only necessary to parse and validate `apisix.yaml` in apisix cli.
+    if default_conf.deployment.config_provider == "yaml" and not ngx then
         local apisix_conf_path = profile:yaml_path("apisix")
```

Comment:
Actually I don't see what the following code does, it just reads the yaml and parses the variables whose values it contains. Other than that, there is no utilization of the values it reads.
So whether we add this extra judgment or not doesn't affect the functionality of using it. I read your PR description and this is just an efficiency difference? I didn't get the need for it.